docs/custom-directive.md

```diff
@@ -0,0 +1,125 @@
+# Custom directive
+
+We might need to customise our schema by decorating parts of it or operations to add new reusable features to these elements.
+To do that, we can use a GraphQL concept called **Directive**.
+
+A GraphQL directive is a special syntax used to provide additional information to the GraphQL execution engine about how to process a query, mutation, or schema definition. 
+Directives can be used to modify the behaviour of fields, arguments, or types in your schema. 
+
+A custom directive is composed of 2 parts:
+- schema definitions
+- transformer
+
+## Schema Definition
+Let's explore the custom directive creation process by creating a directive to redact some fields value, hiding a phone number or an email.
+
+First of all, we must define the schema
+
+```js
+const schema = `
+    directive @redact(find: String) on FIELD_DEFINITION
+    
+    type Document {
+      excerpt: String! @redact(find: "email")
+      text: String! @redact(find: "phone")
+    }
+
+    type Query {
+      documents: [Document] 
+    }`
+```
+
+To define a custom directive, we must use the directive keyword, followed by its name prefixed by a `@`, the arguments (if any), and the locations where it can be applied.
+
+```
+directive @redact(find: String) on FIELD_DEFINITION
+```
+
+According to the graphql specs the directive can be applied in multiple locations. See the list on [the GraphQL spec page](https://spec.graphql.org/October2021/#sec-Type-System.Directives). 
+
+## Transformer
+
+Every directive needs its transformer.
+A transformer is a function that takes an existing schema and applies the modifications to the schema and resolvers.
+
+To simplify the process of creating a transformer, we use the `mapSchema` function from the `@graphql-tools` library.
+
+The `mapSchema` function applies each callback function to the corresponding type definition in the schema, creating a new schema with the modified type definitions. The function also provides access to the field resolvers of each object type, allowing you to alter the behaviour of the fields in the schema.
+
+```js
+const { mapSchema, getDirective, MapperKind } = require('@graphql-tools/utils')
```

Comment:
We should document somewhere what version of graphql-tools this guide refers to `graphql-tools`.

I would also prefer to not use graphql-tools here or show both examples.

Reply:
I agree with Mauro, we have discussed in detail and replicating the features coming from the graphql-tools library for the sake of this example is just not worth it, and quite frankly in general it's not worth it. Nevertheless since this issue comes up quite often, we'll come up with a couple of ideas that we'll open as issues in this repo:

- one specific to this request, a way built-into mercurius (or a plugin) to make it easier to define and attach custom directives
- one generic about creating a new library alternative to graphql-tools with similar features

Reply:
Yes, I have provided a more detailed argument on this issue.